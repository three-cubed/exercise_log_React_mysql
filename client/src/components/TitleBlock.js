import React from 'react';

const TitleBlock = (props) => {
    return (
        <div className='titleBlock'>
            <h2>
                The Exercise Log
            </h2>
            <div className='centred'>
                <button type='button' 
                    style ={{ backgroundColor: props.btnColour}} 
                    className='titleBlockBtn' 
                    onClick={props.onClick}>
                    {props.btnText}
                </button>
            </div>
        </div>
    )
}

export default TitleBlock;
