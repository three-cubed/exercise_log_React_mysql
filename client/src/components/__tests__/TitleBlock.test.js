import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react';
import { toHaveStyle } from '@testing-library/jest-dom'; 
import renderer from 'react-test-renderer';
import TitleBlock from '../TitleBlock.js';

afterEach(cleanup);

describe('TitleBlock component', () => {
    test('renders', () => {
        render(<TitleBlock />);
    });
    test('matches snapshot', () => {
        const tree = renderer.create(<TitleBlock />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('shows title text', () => {
        const newTitleBLock = render(<TitleBlock />);
        expect(newTitleBLock.getByText(/The Exercise Log/));
    });
    test('shows text entered as props.btnText', () => {
        const newTitleBLock = render(<TitleBlock btnText={'someText'} btnColour={'blue'} />);
        expect(newTitleBLock.getByText(/someText/));
    });
    test('shows colour entered as props.btnColour', () => {
        const newTitleBLock = render(<TitleBlock btnText={'someText'} btnColour={'blue'} />);
        expect(newTitleBLock.getByText(/someText/)).toHaveStyle('background-color: blue');
    });
})
