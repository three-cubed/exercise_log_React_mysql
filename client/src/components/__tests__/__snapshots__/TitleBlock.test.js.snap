// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBlock component matches snapshot 1`] = `
<div
  className="titleBlock"
>
  <h2>
    The Exercise Log
  </h2>
  <div
    className="centred"
  >
    <button
      className="titleBlockBtn"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      type="button"
    />
  </div>
</div>
`;
